src/dialog/tests/unit/Dialog.ts

```diff
@@ -19,94 +36,161 @@ registerSuite({
 			role: 'dialog'
 		});
 
-		assert.strictEqual(dialog.properties.key, 'foo');
-		assert.isTrue(dialog.properties.modal);
-		assert.isTrue(dialog.properties.open);
-		assert.strictEqual(dialog.properties.title, 'dialog');
-		assert.isTrue(dialog.properties.underlay);
-		assert.isTrue(dialog.properties.closeable);
-		assert.strictEqual(dialog.properties.role, 'dialog');
+		const expected = v('div', {
+			classes: dialog.classes(css.root)
+		}, [
+			v('div', {
+				classes: dialog.classes(css.underlayVisible, css.underlay),
+				enterAnimation: animations.fadeIn,
+				exitAnimation: animations.fadeOut,
+				key: 'underlay',
+				onclick: dialog.listener,
+				afterCreate: dialog.listener,
+				afterUpdate: dialog.listener
+			}),
+			v('div', {
+				'aria-labelledby': idComparator,
+				classes: dialog.classes(css.main),
+				enterAnimation: animations.fadeIn,
+				exitAnimation: animations.fadeOut,
+				afterCreate: dialog.listener,
+				afterUpdate: dialog.listener,
+				key: 'main',
+				role: 'dialog'
+			}, [
+				v('div', {
+					classes: dialog.classes(css.title),
+					id: <any> idComparator,
+					key: 'title',
+					afterCreate: dialog.listener,
+					afterUpdate: dialog.listener
+				}, [
+					'dialog',
+					v('button', {
+						classes: dialog.classes(css.close),
+						innerHTML: 'close dialog',
+						onclick: dialog.listener
+					})
+				]),
+				v('div', {
+					classes: dialog.classes(css.content),
+					key: 'content',
+					afterCreate: dialog.listener,
+					afterUpdate: dialog.listener
+				}, ['dialog content'])
+			])
+
+		]);
+		dialog.expectRender(expected);
 	},
 
 	'Render correct children'() {
-		const dialog = new Dialog();
-		dialog.__setProperties__({
+		dialog.setProperties({
 			enterAnimation: 'enter',
 			exitAnimation: 'exit',
 			role: 'dialog'
 		});
-		let vnode = <VNode> dialog.__render__();
-		assert.strictEqual(vnode.vnodeSelector, 'div', 'tagname should be div');
-		assert.property(vnode.properties!.classes!, css.root);
+		let vnode = <HNode> dialog.getRender();
 		assert.lengthOf(vnode.children, 0);
 
-		dialog.__setProperties__({
+		dialog.setProperties({
 			open: true,
 			underlay: true,
 			role: 'dialog'
 		});
-		vnode = <VNode> dialog.__render__();
+		vnode = <HNode> dialog.getRender();
 		assert.lengthOf(vnode.children, 2);
 	},
 
 	onRequestClose() {
-		const dialog = new Dialog();
-		dialog.__setProperties__({
+		let called = false;
```

Comment:
might be worth looking into using `Sinon` instead of your `called` variable here. Lends itself to a clearer assertion. ie. `assert.isTrue(requestCloseStub.calledOnce)`.